de/fitting.py

```diff
@@ -76,15 +77,22 @@ def impute(data, fitting=False):
 
         # Fit
         if fitting:
-            xx = runOptim(data, niter=500)
-            w, eta = reshapeParams(xx, data.shape[0])
+            if linear:
+                w, prediction = runFitting(data, U)
+            else:    
+                xx = runOptim(data, niter=500)
+                w, eta = reshapeParams(xx, data.shape[0])
```

Comment:
@AshlynPowell I think the `prediction` used here comes from the same dataset used for training. I think you should use the `model.predict()` on the test set. It probably requires some changes to the `Fitting()` function.